.github/workflows/windows.yml

```diff
@@ -42,52 +42,204 @@ env:
   # clang_sys will search msys path before Program Files\LLVM
   # so we need to override this behaviour until we update clang-sys
   # https://github.com/KyleMayes/clang-sys/issues/150
-  LIBCLANG_PATH: C:\\Program Files\\LLVM\\bin
+  LIBCLANG_PATH: C:\Program Files\LLVM\bin
   RUSTUP_WINDOWS_PATH_ADD_BIN: 1
 
 jobs:
+  # Automatic runner selection for job: build
+  # Runs the underlying job (“workload”) on a self-hosted runner if available,
+  # with the help of a `runner-select` job and a `runner-timeout` job.
+
+  # Selects a self-hosted runner if available, or else a GitHub-hosted runner.
+  # We generate a unique id for the workload, find an idle self-hosted runner
+  # with the given `image:` label that wasn’t already reserved by another
+  # `runner-select` job run, and reserve it with a `reserved-for:<id>` label.
+  runner-select:
+    name: Runner select
+    runs-on: ubuntu-latest
+    outputs:
+      unique-id: ${{ steps.select.outputs.unique_id }}
+      selected-runner-label: ${{ steps.select.outputs.selected_runner_label }}
+      is-self-hosted: ${{ steps.select.outputs.is_self_hosted }}
+    concurrency:
+      group: servo-reserve-self-hosted-runner
+      cancel-in-progress: false
+    permissions: write-all
+    steps:
+      - name: Select and reserve best available runner
+        id: select
+        # Set the variables below to your desired runner images, runner scope
+        # (org or repo), and a token with permission to write to that scope.
+        run: |
+          github_hosted_runner_label=windows-2022
+          self_hosted_image_label=self-hosted-image:windows10
+          self_hosted_runner_scope=/orgs/${{ github.repository_owner }}/actions/runners
+          export GH_TOKEN=${{ secrets.RUNNER_API_TOKEN }}
+
+          fall_back_to_github_hosted() {
+            echo 'Falling back to GitHub-hosted runner'
+            echo "selected_runner_label=$github_hosted_runner_label" | tee -a $GITHUB_OUTPUT
+            echo 'is_self_hosted=false' | tee -a $GITHUB_OUTPUT
+            exit 0
+          }
+
+          # Generate a unique id that allows the workload job to find the runner
+          # we are reserving for it (via runner labels), and allows the timeout
+          # job to find the workload job run (via the job’s friendly name), even
+          # if there are multiple instances in the workflow call tree.
+          random32=$(LC_ALL=C < /dev/urandom tr -dC 0-9a-f 2> /dev/null | head -c 8)
```

Comment:
Could this just use `uuidgen` to generate a true UUID?

Reply:
great idea, i didn’t know about that command! done in 927aad49744c4984ef9d46839f252049c8eaacdf

Reply:
great idea, i didn’t know about that command! and since we no longer dispatch a separate workflow run, humans no longer benefit from the repo and run id. done in bcc52e09aefc7b32424833b414e338cc5f772586